src/ppom-controller.test.ts

```diff
@@ -0,0 +1,202 @@
+import { VERSION_INFO, storageBackendReturningData } from '../test/test-utils';
+import { PPOM } from './ppom';
+import { PPOMController } from './ppom-controller';
+
+Object.defineProperty(globalThis, 'fetch', {
+  writable: true,
+  value: () => undefined,
+});
+
+jest.mock('./ppom.ts', () => ({
+  PPOM: class PPOMClass {
+    #jsonRpcRequest;
+
+    constructor(jsonRpcRequest: any) {
+      this.#jsonRpcRequest = jsonRpcRequest;
+    }
+
+    validateJsonRpc = async () => {
+      return Promise.resolve();
+    };
+
+    free = () => undefined;
+
+    testJsonRPCRequest = () => this.#jsonRpcRequest();
+  },
+  ppomInit: () => undefined,
+}));
+
+const PPOM_VERSION_PATH =
+  'https://storage.googleapis.com/ppom-cdn/ppom_version.json';
+
+const buildFetchSpy = (
+  versionData: any = {
+    status: 200,
+    json: () => VERSION_INFO,
+  },
+  blobData: any = {
+    status: 200,
+    arrayBuffer: () => new ArrayBuffer(123),
+  },
+) => {
+  return jest
+    .spyOn(globalThis, 'fetch' as any)
+    .mockImplementation((url: any) => {
+      if (url === PPOM_VERSION_PATH) {
+        return versionData;
+      }
+      return blobData;
+    });
+};
+
+describe('PPOMController', () => {
+  describe('use', () => {
+    let ppomController: any;
+    beforeEach(() => {
+      ppomController = new PPOMController({
+        storageBackend: storageBackendReturningData,
+        provider: { sendAsync: Promise.resolve() },
+        chainId: '0x1',
+        onNetworkChange: (_callback) => undefined,
+      });
+    });
+
+    it('should be able to invoke use', async () => {
+      buildFetchSpy();
+
+      await ppomController.use(async (ppom: PPOM) => {
```

Comment:
Should we also assert the return value from `use` matches what the callback returns?